src/hastings_index.erl

```diff
@@ -421,6 +473,40 @@ design_doc_to_index_int(Id, Fields, IndexName) ->
     end.
 
 
+verify_index_exists(Options) ->
```

Comment:
I'd look more at removing the checks from hastings here. This function returns a true/false so there's not a lot of reason to try and safely handle all of the various errors directly since they'll all lead to the same result for the function.